contracts/core/VolOracle.sol

```diff
@@ -174,21 +204,79 @@ contract VolOracle is DSMath {
                 .mul(annualizationConstant);
     }
 
+    /**
+     * @notice Returns the TWAP for observation period in USDC terms
+     * @param pool is the Uniswap V3 pool for which we query a TWAP
+     */
+    function assetPriceInUSDC(address pool)
+        public
+        view
+        returns (uint256 price)
+    {
+        IUniswapV3Pool uniPool = IUniswapV3Pool(pool);
+        address token0 = uniPool.token0();
+        address token1 = uniPool.token1();
+        bool isETHPair = token0 == WETH || token1 == WETH;
+
+        if (pool == ethUsdcPool) {
+            // Since this is the ETHUSDC pool, we just use token0, which is USDC as the quoteToken
+            return _twap(pool, token0, token1, 0);
+        } else if (!isETHPair) {
+            // Since we don't know whether we should use token0 or token1 as the quoteToken,
+            // we just default to using token0
+            return _twap(pool, token0, token1, 0);
```

Comment:
so we are assuming !isETHPair means its x-usdc pair?

Reply:
It could be anything, like an OHM-DAI pair, but yes it includes the x-USDC pair